### PR TITLE
(Merge #515 first) initia interaction: function panel and selected function info accordion

### PR DESCRIPTION
**Pull request by poomthiti**

### CODE REVIEW

File: src/lib/pages/interaction/component/common/EmptyState.tsx [hunk 1]
```diff
@@ -41,3 +42,7 @@ export const ModuleEmptyState = ({
     </Flex>
   );
 };
+
+export const NoImageEmptyState = ({ desc }: { desc: string }) => (
+  <ModuleEmptyState h="fit-content" p={4} hasImage={false} description={desc} />
+);
```
Comment (songwongtp) [inline src/lib/pages/interaction/component/common/EmptyState.tsx:48]: why don't we use these NoImage params as a default of params of `ModuleEmptyState`? Only `ModuleEmptyState` is called directly just only 2 times. Also remove default description from `ModuleEmptyState`. Only one location use that default description

File: src/lib/components/module/FunctionCard.tsx [hunk 2]
```diff
@@ -4,47 +4,48 @@ import { Flex, Text } from "@chakra-ui/react";
 import { DotSeparator } from "../DotSeparator";
 import { CustomIcon } from "../icon";
 import { Tooltip } from "lib/components/Tooltip";
-import type { ExposedFunction, Visibility } from "lib/types";
+import type { ExposedFunction } from "lib/types";
+import { getVisibilityIcon } from "lib/utils";
+import { checkAvailability } from "lib/utils/abi";
```
Comment (evilpeach) [inline src/lib/components/module/FunctionCard.tsx:9]: merge it

File: src/lib/services/moduleService.ts [hunk 3]
```diff
@@ -36,30 +41,41 @@ const indexModuleResponse = (module: InternalModule): IndexedModule => {
...
       : getAccountModules(baseEndpoint, address).then((modules) =>
           modules.map((module) => indexModuleResponse(module))
         );
 
   return useQuery(
-    [CELATONE_QUERY_KEYS.ACCOUNT_MODULES, baseEndpoint, address, moduleName],
+    [
+      CELATONE_QUERY_KEYS.ACCOUNT_MODULES,
+      baseEndpoint,
+      address as string,
```
Comment (evilpeach) [inline src/lib/services/moduleService.ts:75]: I think it's no need to convert to type string.

File: src/lib/pages/interaction/component/drawer/selector/SelectorInput.tsx [hunk 4]
```diff
@@ -38,7 +38,10 @@ export const ModuleSelectorInput = ({
 }: ModuleSelectorInputProps) => {
   const [keyword, setKeyword] = useState(selectedAddress.address as string);
   const [error, setError] = useState("");
-  const [addr, moduleName] = useMemo(() => keyword.split("::"), [keyword]);
+  const [addr, moduleName, functionName] = useMemo(
+    () => splitModule(keyword),
+    [keyword]
+  );
```
Comment (evilpeach) [inline src/lib/pages/interaction/component/drawer/selector/SelectorInput.tsx:44]: Should the module or function name be in uppercase or start with a capital letter?

File: src/lib/pages/interaction/component/main/SelectedFunctionCard.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+          variant="ghost-gray"
+          rightIcon={
+            <CustomIcon
+              name="chevron-down"
+              boxSize={3}
+              color="gray.400"
+              transform={expand ? "rotate(180deg)" : "rotate(0)"}
+            />
+          }
+          onClick={() => {
```
Comment (evilpeach) [inline src/lib/pages/interaction/component/main/SelectedFunctionCard.tsx:69]: remove bracket

File: src/lib/pages/interaction/index.tsx [hunk 6]
```diff
@@ -1,38 +1,35 @@
...
+import { useCallback, useState } from "react";
 
-import { CountBadge } from "lib/components/module/CountBadge";
+import { CustomIcon } from "lib/components/icon";
+import { LabelText } from "lib/components/LabelText";
 import PageContainer from "lib/components/PageContainer";
-import { EmptyState } from "lib/components/state";
 import type { IndexedModule } from "lib/services/moduleService";
 import type { ExposedFunction } from "lib/types";
+import { parseJsonABI } from "lib/utils/abi";
```
Comment (evilpeach) [inline src/lib/pages/interaction/index.tsx:16]: 'lib/utils'

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: src/lib/pages/interaction/component/drawer/selector/SelectorInput.tsx (hunk 4 above)
Comment (poomthiti) [reply src/lib/pages/interaction/component/drawer/selector/SelectorInput.tsx:44]: I believe its all in lowercase

**Pull request merged**